tornado-api/src/main/java/uk/ac/manchester/tornado/api/WorkerGrid.java

```diff
@@ -18,23 +18,78 @@
 package uk.ac.manchester.tornado.api;
 
 public interface WorkerGrid {
+
+    /**
+     * Returns the number of parallel dimensions. This
+     * could be 1D, 2D or 3D.
+     * 
+     * @return int
+     */
     int dimension();
```

Comment:
Shall we refactor this method to `numOfDimensions`?

Reply:
I think `dimension` is ok for now.

Reply:
to me the `dimension` could be that you query if you have a 3 dimensional array to get one specific dimension (0, 1, 2). But that's fine for now, and in the java doc it is clear.